### PR TITLE
add orca_submit.sh

### PR DESCRIPTION
**Pull request by Adria777**
add orca_submit.sh, includes :
1. add envs
2. check nfs server
3. deploy pvc
4. copy data to nfs share data
5. submit spark job

### CODE REVIEW

File: scripts/orca_submit.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#env
```
Comment (glorysdj) [inline scripts/orca_submit.sh:1]: we should also add a orca-clean.sh to undeploy pvc and delete related pods

Comment (glorysdj) [reply scripts/orca_submit.sh:1]: so maybe we need to put envs in a seprated orca-env.sh file

File: scripts/orca_submit.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+PVC_STORAGE="100Mi"
+SPARK_K8S_MASTER="k8s://https://127.0.0.1:8443"
+DRIVER_HOST="172.16.0.200"
+DRIVER_PORT="54321"
+AZ_CLIENT_NAME="azclient11"
+K8S_CONFIG="/root/glorysdj/kuberconfig"
+IMAGE="10.239.45.10/arda/hyper-zoo:e2c"
+USER_FOLDER="/root/glorysdj"
+
+COMMEND="cd /zoo/data/e2c-analytics-stack/ref_algos/inspection/train && python train_az_k8s_client.py"
```
Comment (glorysdj) [inline scripts/orca_submit.sh:15]: it's better to cd /zoo/data/ which is mounted dir from nfs
and python e2c-analytics-stack/ref_algos/inspection/train/train_az_k8s_client.py to call specified python program

File: scripts/orca_submit.sh [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+COMMEND1="/opt/analytics-zoo-0.10.0-SNAPSHOT/bin/spark-submit-python-with-zoo.sh \
+--master $SPARK_K8S_MASTER \
+--deploy-mode cluster \
+--name e2c \
+--conf spark.kubernetes.container.image="$IMAGE" \
+--conf spark.kubernetes.executor.volumes.persistentVolumeClaim.nfsvolumeclaim.options.claimName="$PVC_NAME" \
+--conf spark.kubernetes.executor.volumes.persistentVolumeClaim.nfsvolumeclaim.mount.path="/zoo" \
```
Comment (glorysdj) [inline scripts/orca_submit.sh:22]: change MOUNT_PATH as env

File: scripts/orca_submit.sh [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+NFS_SHARE_DATA_PATH="/disk1/nfsdata"
+NFS_CLIENT_PROVISIONER_IMAGE="10.239.45.10/arda/external_storage/nfs-client-provisioner:latest"
+PVC_NAME="persistent-volume-claim"
+PVC_STORAGE="100Mi"
+SPARK_K8S_MASTER="k8s://https://127.0.0.1:8443"
+DRIVER_HOST="172.16.0.200"
+DRIVER_PORT="54321"
+AZ_CLIENT_NAME="azclient11"
+K8S_CONFIG="/root/glorysdj/kuberconfig"
+IMAGE="10.239.45.10/arda/hyper-zoo:e2c"
```
Comment (glorysdj) [inline scripts/orca_submit.sh:12]: IMAGE -> SPARK_K8S_IMAGE

File: scripts/orca_submit.sh (hunk 2 above)
Comment (glorysdj) [inline scripts/orca_submit.sh:15]: can we unify client/cluster mode?

File: scripts/orca_submit.sh [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+#deploy pvc
+sed -n "s/TMP_NFS_SERVER_VALUE/$NFS_SERVER_VALUE/g" deploy-nfs/deployment.yaml.template | tee deploy-nfs/deployment.yaml
+sed -i "s#TMP_NFS_SHARE_DATA_PATH#$NFS_SHARE_DATA_PATH#g" deploy-nfs/deployment.yaml
+sed -i "s#TMP_NFS_CLIENT_PROVISIONER_IMAGE#$NFS_CLIENT_PROVISIONER_IMAGE#g" deploy-nfs/deployment.yaml
+sed -n "s/TMP_PVC_NAME/$PVC_NAME/g" deploy-nfs/persistent-volume-claim.yaml.template | tee deploy-nfs/persistent-volume-claim.yaml
+sed -i "s/TMP_PVC_STORAGE/$PVC_STORAGE/g" deploy-nfs/persistent-volume-claim.yaml
+
+kubectl create -f deploy-nfs/deployment.yaml
+kubectl create -f deploy-nfs/class.yaml
+kubectl create -f deploy-nfs/persistent-volume-claim.yaml
```
Comment (glorysdj) [inline scripts/orca_submit.sh:56]: we should clear them after the job

File: scripts/orca_submit.sh [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+#copy to nfsdata
+var1=`ls $NFS_SHARE_DATA_PATH | grep "$PVC_NAME" | grep -v "archived"`
+echo $var1
+cp -r $USER_FOLDER/data $NFS_SHARE_DATA_PATH/$var1
+mkdir $NFS_SHARE_DATA_PATH/$var1/logs
+
+#run image
```
Comment (glorysdj) [inline scripts/orca_submit.sh:64]: to run client container

File: scripts/orca_submit.sh [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+sed -i "s/TMP_PVC_STORAGE/$PVC_STORAGE/g" deploy-nfs/persistent-volume-claim.yaml
+
+kubectl create -f deploy-nfs/deployment.yaml
+kubectl create -f deploy-nfs/class.yaml
+kubectl create -f deploy-nfs/persistent-volume-claim.yaml
+
+#copy to nfsdata
+var1=`ls $NFS_SHARE_DATA_PATH | grep "$PVC_NAME" | grep -v "archived"`
+echo $var1
+cp -r $USER_FOLDER/data $NFS_SHARE_DATA_PATH/$var1
```
Comment (glorysdj) [inline scripts/orca_submit.sh:61]: only copy $USER_FOLDER, and better to name is as USER_LOCAL_STORAGE

File: scripts/orca_submit.sh [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+NFS_CLIENT_PROVISIONER_IMAGE="10.239.45.10/arda/external_storage/nfs-client-provisioner:latest"
+PVC_NAME="persistent-volume-claim"
+PVC_STORAGE="100Mi"
+SPARK_K8S_MASTER="k8s://https://127.0.0.1:8443"
+DRIVER_HOST="172.16.0.200"
+DRIVER_PORT="54321"
+AZ_CLIENT_NAME="azclient11"
+K8S_CONFIG="/root/glorysdj/kuberconfig"
+IMAGE="10.239.45.10/arda/hyper-zoo:e2c"
+USER_FOLDER="/root/glorysdj"
```
Comment (glorysdj) [inline scripts/orca_submit.sh:13]: USER_FOLDER -> USER_LOCAL_STORAGE 
add also add comments 
#put the user data and program here, they will be upload to shared storage to each spark driver and executors.

File: scripts/orca_submit.sh [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+var1=`ls $NFS_SHARE_DATA_PATH | grep "$PVC_NAME" | grep -v "archived"`
+echo $var1
+cp -r $USER_FOLDER/data $NFS_SHARE_DATA_PATH/$var1
+mkdir $NFS_SHARE_DATA_PATH/$var1/logs
+
+#run image
+docker pull $IMAGE
+docker run -itd --net=host --name $AZ_CLIENT_NAME \
+    -v $K8S_CONFIG:/root/.kube/config \
+    -v $USER_FOLDER:/zoo \
```
Comment (glorysdj) [inline scripts/orca_submit.sh:68]: change /zoo to $MOUNT_PATH, the folder mount to executor and driver can be unified

File: scripts/orca_submit.sh [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    -e RUNTIME_SPARK_MASTER=$SPARK_K8S_MASTER \
+    -e RUNTIME_K8S_SERVICE_ACCOUNT=spark \
```
Comment (glorysdj) [inline scripts/orca_submit.sh:70]: RUNTIME_K8S_SERVICE_ACCOUNT RUNTIME_EXECUTOR_INSTANCES RUNTIME_EXECUTOR_CORES RUNTIME_EXECUTOR_MEMORY RUNTIME_TOTAL_EXECUTOR_CORES RUNTIME_DRIVER_CORES RUNTIME_DRIVER_MEMORY
should add as envs

File: scripts/orca_submit.sh [hunk 11]
```diff
@@ -0,0 +1,83 @@
+#env
+NFS_SERVER_VALUE="172.16.0.200"
```
Comment (glorysdj) [inline scripts/orca_submit.sh:2]: better to NFS_SERVER

File: scripts/orca_submit.sh (hunk 8 above)
Comment (glorysdj) [reply scripts/orca_submit.sh:13]: envs can be seprated to 2 parts:
NFS_SERVER
NFS_SHARE_DATA_PATH
PVC_NAME
PVC_STORAGE
USER_LOCAL_STORAGE
USER_COMMAND

and others

**Comment by Adria777** [comment]
create a new orca-submit request, so this old one need to be close.

**Pull request closed without merging**